src/envoy/utils/stats.cc

```diff
@@ -53,47 +53,40 @@ void MixerStatsObject::OnTimer() {
   timer_->enableTimer(std::chrono::milliseconds(stats_update_interval_));
 }
 
+#define CHECK_AND_UPDATE_STATS(NAME)                   \
+  if (new_stats.NAME > old_stats_.NAME) {              \
+    stats_.NAME.add(new_stats.NAME - old_stats_.NAME); \
+  }
+
 void MixerStatsObject::CheckAndUpdateStats(
     const ::istio::mixerclient::Statistics& new_stats) {
-  if (new_stats.total_check_calls > old_stats_.total_check_calls) {
-    stats_.total_check_calls_.add(new_stats.total_check_calls -
-                                  old_stats_.total_check_calls);
-  }
-  if (new_stats.total_remote_check_calls >
-      old_stats_.total_remote_check_calls) {
-    stats_.total_remote_check_calls_.add(new_stats.total_remote_check_calls -
-                                         old_stats_.total_remote_check_calls);
-  }
-  if (new_stats.total_blocking_remote_check_calls >
-      old_stats_.total_blocking_remote_check_calls) {
-    stats_.total_blocking_remote_check_calls_.add(
-        new_stats.total_blocking_remote_check_calls -
-        old_stats_.total_blocking_remote_check_calls);
-  }
-  if (new_stats.total_quota_calls > old_stats_.total_quota_calls) {
-    stats_.total_quota_calls_.add(new_stats.total_quota_calls -
-                                  old_stats_.total_quota_calls);
-  }
-  if (new_stats.total_remote_quota_calls >
-      old_stats_.total_remote_quota_calls) {
-    stats_.total_remote_quota_calls_.add(new_stats.total_remote_quota_calls -
-                                         old_stats_.total_remote_quota_calls);
-  }
-  if (new_stats.total_blocking_remote_quota_calls >
-      old_stats_.total_blocking_remote_quota_calls) {
-    stats_.total_blocking_remote_quota_calls_.add(
-        new_stats.total_blocking_remote_quota_calls -
-        old_stats_.total_blocking_remote_quota_calls);
-  }
-  if (new_stats.total_report_calls > old_stats_.total_report_calls) {
-    stats_.total_report_calls_.add(new_stats.total_report_calls -
-                                   old_stats_.total_report_calls);
-  }
-  if (new_stats.total_remote_report_calls >
-      old_stats_.total_remote_report_calls) {
-    stats_.total_remote_report_calls_.add(new_stats.total_remote_report_calls -
-                                          old_stats_.total_remote_report_calls);
-  }
+  CHECK_AND_UPDATE_STATS(total_check_calls_);
```

Comment:
Do we need trailing underscore "_". It should be new_stats.total_check_calls instead of new_stats.total_check_calls_, no?

Reply:
I see, you have updated stats names with "_". Please ignore my comment.

Reply:
@JimmyCYJ thank you for jumping on this PR!  appreciate it :)